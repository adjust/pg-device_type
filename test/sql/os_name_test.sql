BEGIN;
-- device_type should pass valid device_type;
-- ./spec/os_name_spec.rb:8;
CREATE EXTENSION device_type;
SELECT 'bot'::device_type;
SELECT 'console'::device_type;
SELECT 'ipod'::device_type;
SELECT 'mac'::device_type;
SELECT 'pc'::device_type;
SELECT 'phone'::device_type;
SELECT 'server'::device_type;
SELECT 'simulator'::device_type;
SELECT 'tablet'::device_type;
SELECT 'tv'::device_type;
SELECT 'unknown'::device_type;
ROLLBACK;
BEGIN;
-- device_type should rais an exception on invalid device_type;
-- ./spec/os_name_spec.rb:26;
CREATE EXTENSION device_type;
SELECT 'kermit'::device_type;
ROLLBACK;
BEGIN;
-- device_type should rais an exception on invalid device_type;
-- ./spec/os_name_spec.rb:30;
CREATE EXTENSION device_type;
SELECT 'tb'::device_type;
ROLLBACK;
