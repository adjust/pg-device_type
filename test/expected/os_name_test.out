BEGIN;
-- device_type should pass valid device_type;
-- ./spec/os_name_spec.rb:8;
CREATE EXTENSION device_type;
SELECT 'bot'::device_type;
 device_type 
-------------
 bot
(1 row)

SELECT 'console'::device_type;
 device_type 
-------------
 console
(1 row)

SELECT 'ipod'::device_type;
 device_type 
-------------
 ipod
(1 row)

SELECT 'mac'::device_type;
 device_type 
-------------
 mac
(1 row)

SELECT 'pc'::device_type;
 device_type 
-------------
 pc
(1 row)

SELECT 'phone'::device_type;
 device_type 
-------------
 phone
(1 row)

SELECT 'server'::device_type;
 device_type 
-------------
 server
(1 row)

SELECT 'simulator'::device_type;
 device_type 
-------------
 simulator
(1 row)

SELECT 'tablet'::device_type;
 device_type 
-------------
 tablet
(1 row)

SELECT 'tv'::device_type;
 device_type 
-------------
 tv
(1 row)

SELECT 'unknown'::device_type;
 device_type 
-------------
 unknown
(1 row)

ROLLBACK;
BEGIN;
-- device_type should rais an exception on invalid device_type;
-- ./spec/os_name_spec.rb:26;
CREATE EXTENSION device_type;
SELECT 'kermit'::device_type;
ERROR:  unknown input device_type: kermit
LINE 1: SELECT 'kermit'::device_type;
               ^
ROLLBACK;
